### PR TITLE
Dump load

### PR DESCRIPTION
**Pull request by rohe**
The dump/load methods used for persistence storage of state was not working well in the general case.
This PR fixes that

### CODE REVIEW

**Comment by rohe** [comment]
@lionick and @peppelinux I really would like to have your reviews.

File: src/idpyoidc/impexp.py [hunk 1]
```diff
@@ -180,6 +188,18 @@ def load(self, item: dict, init_args: Optional[dict] = None, load_args: Optional
...
+            for param, target in {"upstream_get": "unit_get", "conf": "conf",
+                                  "token_handler_args": "token_handler_args"}.items():
+                target_val = getattr(self, target, None)
+                if target_val:
+                    if param in _cls_init_args and param not in _kwargs:
+                        if _kwargs["init_args"] is None:
+                            _kwargs["init_args"] = {param: target_val}
+                        else:
+                            _kwargs["init_args"][param] = target_val
+
```
Comment (peppelinux) [inline src/idpyoidc/impexp.py]:
```suggestion
                if not target_val:
                    continue
                    
                if param in _cls_init_args and param not in _kwargs:
                    if not _kwargs["init_args"]:
                        _kwargs["init_args"] = {param: target_val}
                    else:
                        _kwargs["init_args"][param] = target_val

```

**Review by peppelinux — CHANGES_REQUESTED** [review]
I didn't have time to test it with satosa-oidcop but it seems formally good, I just have added a quick suggestion to reduce the hadoken code

**Review by lionick — APPROVED** [review]
Running my tests at oidc frontend, I can confirm that everything works as expected.

**Comment by rohe** [comment]
Completely baffled. Checks fail on a test that is not part of the package ?!

**Pull request merged**